api/v1alpha1/promotion_types.go

```diff
@@ -109,6 +109,27 @@ type PromotionVariable struct {
 	Value string `json:"value" protobuf:"bytes,2,opt,name=value"`
 }
 
+// PromotionRetry describes the retry policy for a PromotionStep.
+type PromotionRetry struct {
```

Comment:
Given #2972, I wonder if we should name this `PromotionStepRetry` or even `PromotionStepRetryPolicy` as I think we will probably also end up with Promotion-level retry policies in addition to step-level.

Reply:
Non-blocking.

Reply:
Renamed it to `PromotionStepRetry` to reserve the policy version for e.g. some enum that describes how it should be retried.